File: 2676-find-the-score-of-all-prefixes-of-an-array/find-the-score-of-all-prefixes-of-an-array.java
class Solution {
    public long[] findPrefixScore(int[] nums) {
        long sum1 = 0 ;
        long max = Long.MIN_VALUE;
        long ans [] = new long [nums.length];
        for(int i = 0; i < nums.length; i++){
            max = Math.max(max, nums[i]);
            sum1 += max + nums[i];
            ans[i] = sum1;
        }
        return ans;
    }
}

File: 2676-find-the-score-of-all-prefixes-of-an-array/find-the-score-of-all-prefixes-of-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findPrefixScore = function (nums) {
    let sum1 = 0;
    let max = 0;
    let ans = [];
    for (let i = 0; i < nums.length; i++) {
        max = Math.max(max, nums[i]);
        sum1 += max + nums[i];
        ans.push(sum1);
    }
    return ans;
};